diff --git a/deps/undici/src/lib/web/fetch/util.js b/deps/undici/src/lib/web/fetch/util.js
index fe63cb3a9b07e57198f08d6237d2b7b39bfd057d..41fa934bbcaf62d29c605906f55497de69522b15 100644
--- a/deps/undici/src/lib/web/fetch/util.js
+++ b/deps/undici/src/lib/web/fetch/util.js
@@ -237,13 +237,13 @@ function appendFetchMetadata (httpRequest) {
   //  TODO
 
   //  2. Let header be a Structured Header whose value is a token.
-  let header = null
+  // let header = null
 
   //  3. Set header’s value to r’s mode.
-  header = httpRequest.mode
+  // header = httpRequest.mode
 
   //  4. Set a structured field value `Sec-Fetch-Mode`/header in r’s header list.
-  httpRequest.headersList.set('sec-fetch-mode', header, true)
+  // httpRequest.headersList.set('sec-fetch-mode', header, true)
 
   //  https://w3c.github.io/webappsec-fetch-metadata/#sec-fetch-site-header
   //  TODO
diff --git a/deps/undici/undici.js b/deps/undici/undici.js
index 22ec3444e6ec307dde88be3be4a63e20d2ee0133..2e3297550204e39ee99d3d059bd235de62d9570e 100644
--- a/deps/undici/undici.js
+++ b/deps/undici/undici.js
@@ -5389,9 +5389,9 @@ var require_util2 = __commonJS({
     }
     __name(TAOCheck, "TAOCheck");
     function appendFetchMetadata(httpRequest) {
-      let header = null;
-      header = httpRequest.mode;
-      httpRequest.headersList.set("sec-fetch-mode", header, true);
+      // let header = null;
+      // header = httpRequest.mode;
+      // httpRequest.headersList.set("sec-fetch-mode", header, true);
     }
     __name(appendFetchMetadata, "appendFetchMetadata");
     function appendRequestOriginHeader(request) {
