diff --git a/deps/undici/src/lib/web/fetch/util.js b/deps/undici/src/lib/web/fetch/util.js
index 775ef296db26226c45359c2076b22740f6bea426..aac7af2cfad6bef1ce8bdc569d26f9b9952cca53 100644
--- a/deps/undici/src/lib/web/fetch/util.js
+++ b/deps/undici/src/lib/web/fetch/util.js
@@ -237,13 +237,13 @@ function appendFetchMetadata (httpRequest) {
   //  TODO
 
   //  2. Let header be a Structured Header whose value is a token.
-  let header = null
+  // let header = null
 
   //  3. Set header’s value to r’s mode.
-  header = httpRequest.mode
+  // header = httpRequest.mode
 
   //  4. Set a structured field value `Sec-Fetch-Mode`/header in r’s header list.
-  httpRequest.headersList.set('sec-fetch-mode', header, true)
+  // httpRequest.headersList.set('sec-fetch-mode', header, true)
 
   //  https://w3c.github.io/webappsec-fetch-metadata/#sec-fetch-site-header
   //  TODO
diff --git a/deps/undici/undici.js b/deps/undici/undici.js
index 19915b8029af9fd6a50837eb55c168585a0fba48..8a2f9dd1d9c74d2b3c0287d93ecc2c71afef9c39 100644
--- a/deps/undici/undici.js
+++ b/deps/undici/undici.js
@@ -5398,9 +5398,9 @@ var require_util2 = __commonJS({
     }
     __name(TAOCheck, "TAOCheck");
     function appendFetchMetadata(httpRequest) {
-      let header = null;
-      header = httpRequest.mode;
-      httpRequest.headersList.set("sec-fetch-mode", header, true);
+      // let header = null;
+      // header = httpRequest.mode;
+      // httpRequest.headersList.set("sec-fetch-mode", header, true);
     }
     __name(appendFetchMetadata, "appendFetchMetadata");
     function appendRequestOriginHeader(request) {
