diff --git a/deps/uv/src/threadpool.c b/deps/uv/src/threadpool.c
index 2a129a5d4a581bcd67599101b666eee3f9a88218..e9c3fc7b95ebfa2b97f497175a9c0b6d30c182f6 100644
--- a/deps/uv/src/threadpool.c
+++ b/deps/uv/src/threadpool.c
@@ -201,7 +201,9 @@ static void init_threads(void) {
 
   uv_sem_t sem;
 
-  nthreads = ARRAY_SIZE(default_threads);
+  nthreads = uv_available_parallelism();
+  if (nthreads < ARRAY_SIZE(default_threads))
+    nthreads = ARRAY_SIZE(default_threads);
 
   buflen = ARRAY_SIZE(buf);
   err = uv_os_getenv("UV_THREADPOOL_SIZE", buf, &buflen);
